flytekit/extras/accelerators.py

```diff
@@ -0,0 +1,87 @@
+import abc
+import copy
+import enum
+from typing import Generic, Optional, Type, TypeVar
+
+from flyteidl.core import tasks_pb2
+
+T = TypeVar("T")
+ACC = TypeVar("ACC", bound="BaseAccelerator")
+MIG = TypeVar("MIG", bound="MultiInstanceGPUAccelerator")
+PART = TypeVar("PART", bound=enum.Enum)
+
+
+class BaseAccelerator(abc.ABC, Generic[T]):
+    @abc.abstractmethod
+    def to_flyte_idl(self) -> T:
+        ...
+
+
+class GPUAccelerator(BaseAccelerator):
+    def __init__(self, device: str) -> None:
+        self._device = device
+
+    def to_flyte_idl(self) -> tasks_pb2.GPUAccelerator:
+        return tasks_pb2.GPUAccelerator(device=self._device)
+
+
+NvidiaA10G = GPUAccelerator("nvidia-a10g")
+NvidiaL4 = GPUAccelerator("nvidia-l4-vws")
+NvidiaTeslaK80 = GPUAccelerator("nvidia-tesla-k80")
+NvidiaTeslaM60 = GPUAccelerator("nvidia-tesla-m60")
+NvidiaTeslaP4 = GPUAccelerator("nvidia-tesla-p4")
+NvidiaTeslaP100 = GPUAccelerator("nvidia-tesla-p100")
+NvidiaTeslaT4 = GPUAccelerator("nvidia-tesla-t4")
+NvidiaTeslaV100 = GPUAccelerator("nvidia-tesla-v100")
+
+
+class MultiInstanceGPUAccelerator(BaseAccelerator, Generic[PART]):
+    _partition_size: Optional[PART]
+
+    def __init__(self, device: str, partition_sizes: Type[PART]) -> None:
+        self._device = device
+        self._partition_sizes = partition_sizes
+
+    @property
+    def partition_sizes(self) -> Type[PART]:
+        return self._partition_sizes
+
+    def with_partition_size(self: MIG, partition_size: Optional[PART]) -> MIG:
```

Comment:
can we change the typehint for `partition_size` to be `PART`? If the user is going through the trouble of caling `with_partition_size` it's reasonable to assume that they have a real value. wdyt?

Reply:
`.with_partition_size(None)` is still valid, for explicitly requesting an unpartitioned GPU.